packages/functional-tests/lib/accountManager.ts

```diff
@@ -0,0 +1,145 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import crypto from 'crypto';
+import { Credentials } from './targets';
+import { BaseTarget } from './targets/base';
+
+const BLOCKED_EMAIL_PREFIX = 'blocked{id}';
+const BOUNCED_EMAIL_PREFIX = 'bounced{id}';
+const FORCED_PWD_EMAIL_PREFIX = 'forcepwdchange{id}';
+const SIGNIN_EMAIL_PREFIX = 'signin{id}';
+const SIGNUP_REACT_EMAIL_PREFIX = 'signup_react{id}';
+const SYNC_EMAIL_PREFIX = 'sync{id}';
+
+type EmailPrefix =
+  | typeof BLOCKED_EMAIL_PREFIX
+  | typeof BOUNCED_EMAIL_PREFIX
+  | typeof FORCED_PWD_EMAIL_PREFIX
+  | typeof SIGNIN_EMAIL_PREFIX
+  | typeof SIGNUP_REACT_EMAIL_PREFIX
+  | typeof SYNC_EMAIL_PREFIX;
+
+type Account = {
+  email: string;
+  password: string;
+};
+
+export class AccountManager {
+  accounts: (Account | Credentials)[];
+  private target: BaseTarget;
+
+  constructor(target: BaseTarget) {
+    this.target = target;
+    this.accounts = [];
+  }
+
+  generateBlockedEmail(): string {
+    return this.generateEmail(BLOCKED_EMAIL_PREFIX);
+  }
+
+  generateSyncEmail(): string {
+    return this.generateEmail(SYNC_EMAIL_PREFIX);
+  }
+
+  generateEmail(prefixTemplate: EmailPrefix = SIGNIN_EMAIL_PREFIX): string {
+    const prefix = prefixTemplate.replace(
+      '{id}',
+      crypto.randomBytes(8).toString('hex')
+    );
+    return `${prefix}@restmail.net`;
+  }
+
+  generatePassword(): string {
+    return `${crypto.randomBytes(10).toString('hex')}`;
+  }
+
+  generateBouncedAccount(): Account {
+    return this.generateAccount(BOUNCED_EMAIL_PREFIX);
+  }
+
+  generateSignupReactAccount(): Account {
+    return this.generateAccount(SIGNUP_REACT_EMAIL_PREFIX);
+  }
+
+  generateSyncAccount(): Account {
+    return this.generateAccount(SYNC_EMAIL_PREFIX);
+  }
+
+  generateAccount(prefix: EmailPrefix = SIGNIN_EMAIL_PREFIX): Account {
```

Comment:
nit: `generateAccount` reads to me that it is creating the account behind the scenes. Something like `generateAccountDetails`might be clearer? I think we could add some jsdoc on some of these functions to help clear up any confusion.

Reply:
changes applied (see https://github.com/mozilla/fxa/pull/16739/commits/35b8556e238881bd046e784426ec270aa59a2243)